clang/test/Driver/sycl-device-lib.cpp

```diff
@@ -128,7 +128,7 @@
 
 /// ###########################################################################
 /// test llvm-link behavior for fno-sycl-device-lib
-// RUN: %clangxx -fsycl -fno-sycl-device-lib=all %s -### 2>&1 \
+// RUN: %clangxx -fsycl -fno-sycl-dead-args-optimization -fno-sycl-device-lib=all %s -### 2>&1 \
```

Comment:
What exactly is the backstory for the no-device-libs mode?

Reply:
The device libraries are pulled in by default now, which wasn't the case initially.  So instead of modifying the large number of tests that were created without the device libraries in mind, we added the `no-device-lib` to reduce test changes.